### DISCUSSION TITLE
useFieldArray fields not updating

### DISCUSSION THREAD

**Discussion by eugenegordeiev** [react-hook-form/react-hook-form#2473] [Q&A]
I'm trying to dynamically build a different form based on user selection.  I'm using conditional rendering but fields are not updating.  So I have to use watch.  I'm wondering is this the correct way of doing things?

Also, when using useFieldArray, how can I watch a particular field instead of watching the entire array of fields?

See URL for example:
https://codesandbox.io/s/react-hook-form-usefieldarray-template-g7zzm?file=/src/index.js:107-120

**Comment by bluebill1049** [comment]
try to use CSS display property for toggle inputs.
